Update Dependencies (REUPLOAD)

CDAGaming opened the pull request:
Uglify-JS: 2.8.22 > 3.0.11
QUnitJS: 2.3.1 > 2.3.2
PostCSS-CLI: 3.1.1 > 4.0.0
Clean-CSS-CLI: 4.0.12 > 4.1.3
PostCSS-FlexBugs-Fixes: 2.1.1 > 3.0.0
Babel-Eslint: 7.2.2 > 7.2.3
AutoPrefixer: 6.7.7 > 7.1.1
Node-Sass: 4.5.2 > 4.5.3

CDAGaming committed:
Update Dependencies (REUPLOAD)

Uglify-JS: 2.8.22 > 3.0.3
QUnitJS: 2.3.1 > 2.3.2
PostCSS-CLI: 3.1.1 > 4.0.0
Clean-CSS-CLI: 4.0.12 > 4.1.0
Babel-Eslint: 7.2.2 > 7.2.3
AutoPrefixer: 6.7.7 > 7.0.1

wolfy1339 commented:
The Autoprefixer upgrade is a major one, relevent notes from the Changelog:

## 7.0.1
* Fix Autoprefixer for old JS runtimes.

## 7.0 “Coelestem adspicit lucem”
* Remove node.js 0.12 support.
* Use PostCSS 6.0.
* Use Browserslist 2.
* Use `caniuse-lite` instead of `caniuse-db` (by Ben Briggs).
* Use `^` for Browserslist dependencies, instead of `~`.
* Disable Grid Layout prefixes for IE by default.
* Fix `-ms-grid-column-align`.
```

CDAGaming commented:
Seems Odd that on the latest commit it's referencing TWO different version of JQuery. In config.yml also, that jQuery file CDN can be adjusted to cdnjs, while I'll most likely add to this PR later today

CDAGaming committed:
Merge branch 'v4-dev' into v4-dev

mdo commented:
This will need some reviewing and testing before we accept. Can anyone confirm all the dist files build properly and there are no errors?

CDAGaming committed:
Merge branch 'v4-dev' into v4-dev

CDAGaming commented:
May Need to recheck these dependencies, as some may have updated once more.

CDAGaming commented:
Yup, some deps need More Updating, one moment

CDAGaming committed:
Updates More Dependencies

Autoprefixer: v7.0.1 > v7.1.1
Clean-CSS-CLI: v4.1.0 > v4.1.3
Node-Sass: v4.5.2 > v4.5.3
PostCSS-FlexBugs-Fixes: v2.1.1 > v3.0.0
Uglify-JS: v3.0.3 > v3.0.11
Signed-off-by: CDAGaming <cstack2011@yahoo.com>

CDAGaming committed:
Merge branch 'v4-dev' of https://github.com/CDAGaming/bootstrap into v4-dev

CDAGaming committed:
Update Gem Bundler Dependency

Signed-off-by: CDAGaming <cstack2011@yahoo.com>

CDAGaming commented:
Local Environment and Travis Building is Successful, and Bundler Version Dependency is now also Updated

CDAGaming commented:
Just Realized we are now using popper, which has a LOT of outdated dependencies, RIP 👍

Johann-S commented:
It seems @bardiharborow made a test in our `deps` branch (see : https://github.com/twbs/bootstrap/commit/eb0130ad9bdaf175d11d75e76ec04d3326947893)

and Travis build failed https://travis-ci.org/twbs/bootstrap/builds/231071098

CDAGaming committed:
Update Gemfile.lock Dependency

Signed-off-by: CDAGaming <cstack2011@yahoo.com>

CDAGaming commented:
Odd @Johann-S ,as in this PR all builds are reporting successfull, and the main errors im seeing are in relation to QUnit, ill pull the changes done in deps branch and see whats up with it

CDAGaming commented:
So, looking into the changes, the deps referenced still seem to be a bit outdated. I've also noticed that 'docs/assets/js/vendor/jquery-slim.min.js' was removed, but it appears that the errors in Travis is still referencing this file. Seems at this time, that that could be a probable cause of those errors. @bardiharborow You may reference the newer commits on this PR, and try to add back 'docs/assets/js/vendor/jquery-slim.min.js', which might fix travis error

CDAGaming committed:
Update QUnit

Signed-off-by: CDAGaming <cstack2011@yahoo.com>

CDAGaming committed:
Merge branch 'v4-dev' into v4-dev

CDAGaming commented:
As you can see, Updated QUnit with No issues. Most likely error cause was that deleted file most likely

Johann-S reviewed (approved):
LGTM 👍

CDAGaming committed:
Merge branch 'v4-dev' into v4-dev

CDAGaming committed:
Update Uglify-JS to v3.0.13

Signed-off-by: CDAGaming <cstack2011@yahoo.com>

wolfy1339 commented on js/tests/vendor/qunit.css:
```diff
@@ -433,4 +433,4 @@
 	width: 1000px;
 	height: 1000px;
-}
+}
```
Please re-add the final new-line

CDAGaming replied:
It's there. was just an invisible diff, and shouldnt need to be worried about

mdo committed:
Merge branch 'v4-dev' into v4-dev
